pkg/streamingpromql/types/hpoint_ring_buffer.go

```diff
@@ -172,6 +173,11 @@ func (b *HPointRingBuffer) NextPoint() (*promql.HPoint, error) {
 		copy(newSlice, b.points[b.firstIndex:])
 		copy(newSlice[pointsAtEnd:], b.points[:b.firstIndex])
 
+		// We must clear b.points before returning it to the pool, as the current query could continue using the
+		// FloatHistogram instances it contains a reference to, but a later user of b.points may otherwise reuse
+		// those instances instead of creating new FloatHistograms.
+		clear(b.points)
```

Comment:
Just to help my understanding here.. is this because a future query may end up with the released slice pool?

Reply:
Yep, exactly.